File: WCFDemo.Host/JsonData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;

namespace WCFDemo.Host
{
    [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
    [DataContract]
    public class JsonData
    {
        [DataMember(Name = "user")]
        public User User { get; set; }

        [DataMember(Name = "projects")]
        public Project[] Projects { get; set; }
    }

    [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
    [DataContract]
    public class User
    {
        [DataMember(Name = "id")]
        public int Id { get; set; }

        [DataMember(Name = "name")]
        public string Name { get; set; }

        [DataMember(Name = "age")]
        public int Age { get; set; }
    }

    [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
    [DataContract]
    public class Project
    {
        [DataMember(Name = "id")]
        public int Id { get; set; }

        [DataMember(Name = "name")]
        public string Name { get; set; }

        [DataMember(Name = "isActive")]
        public bool IsActive { get; set; }
    }
}

File: WCFDemo.HostVB/JsonData.vb
Imports Newtonsoft.Json
Imports Newtonsoft.Json.Serialization

<JsonObject(NamingStrategyType:=GetType(CamelCaseNamingStrategy))> _
Public Class JsonData
    Private _user As User
    Private _projects As Project()

    Public Property User() As User
        Get
            Return _user
        End Get
        Set(ByVal value As User)
            _user = value
        End Set
    End Property

    Public Property Projects() As Project()
        Get
            Return _projects
        End Get
        Set(ByVal value As Project())
            _projects = value
        End Set
    End Property
End Class

<JsonObject(NamingStrategyType:=GetType(CamelCaseNamingStrategy))> _
Public Class User
    Private _id As Integer
    Private _name As String
    Private _age As Integer

    Public Property Id() As Integer
        Get
            Return _id
        End Get
        Set(ByVal value As Integer)
            _id = value
        End Set
    End Property

    Public Property Name() As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            _name = value
        End Set
    End Property

    Public Property Age() As Integer
        Get
            Return _age
        End Get
        Set(ByVal value As Integer)
            _age = value
        End Set
    End Property
End Class

<JsonObject(NamingStrategyType:=GetType(CamelCaseNamingStrategy))> _
Public Class Project
    Private _id As Integer
    Private _name As String
    Private _isActive As Boolean

    Public Property Id() As Integer
        Get
            Return _id
        End Get
        Set(ByVal value As Integer)
            _id = value
        End Set
    End Property

    Public Property Name() As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            _name = value
        End Set
    End Property

    Public Property IsActive() As Boolean
        Get
            Return _isActive
        End Get
        Set(ByVal value As Boolean)
            _isActive = value
        End Set
    End Property
End Class
